### PR TITLE
Switch costs to `Vector<std::unique_ptr<LayoutData<Real>>>`

### PR DESCRIPTION
**Pull request by mrowan137**
This PR (dependency: #955) switches the `costs` used in load balancing from `Vector<std::unique_ptr<Vector<Real>>>` to `Vector<std::unique_ptr<LayoutData<Real>>>`.  `amrex::LayoutData` is a 'one-item-per-box' structure and is more natural than using a `Vector` to keep track of costs, since we only care about the cost corresponding to each box.

### CODE REVIEW

File: Source/Initialization/WarpXInitData.cpp [hunk 1]
```diff
@@ -416,9 +416,9 @@ WarpX::InitLevelData (int lev, Real /*time*/)
     }
 
     if (costs[lev]) {
-        std::fill((*costs[lev]).begin(),
-                  (*costs[lev]).end(),
-                  0.0);
+        for (int i : costs[lev]->IndexArray()) {
+            (*costs[lev])[i] = 0.0;
```
Comment (ax3l) [inline Source/Initialization/WarpXInitData.cpp:420]: ah, too bad this does not have an `->at()` method :)
```suggestion
            costs[lev]->at(i) = 0.0;
```

File: Source/Initialization/WarpXInitData.cpp [hunk 2]
```diff
@@ -416,9 +416,9 @@ WarpX::InitLevelData (int lev, Real /*time*/)
...
+        }
```
Comment (ax3l) [inline Source/Initialization/WarpXInitData.cpp:421]: Does this object support
```suggestion
        for (auto& c : costs[lev]) {
            c = 0.0;
        }
```
?

**Comment by ax3l** [comment]
@mrowan137 LGTM, thank you! :) Just needs to resolve the merge conflict with `WarpX.H` in `master`, please.

File: Source/Initialization/WarpXInitData.cpp (hunk 2 above)
Comment (mrowan137) [reply Source/Initialization/WarpXInitData.cpp:425]: I think it's not supported, unfortunately.  Maybe worth an AMReX PR?

**Pull request merged**